custom_components/tuya_local/devices/xoca_dac2121c_energy_meter.yaml

```diff
@@ -0,0 +1,158 @@
+name: XOCA DAC2121C BI Energy Meter
+products:
+  - id: opdhepy0p7eyeurz
+    name: DAC-2121C BI
+primary_entity:
+  entity: switch
+  dps:
+    - id: 16
+      name: switch
+      type: boolean
+    - id: 6
+      name: phase_a
+      type: base64
+      optional: true
+      force: true
+secondary_entities:
+  - entity: sensor
+    name: Total Forward Energy
+    class: energy
+    # category: diagnostic
+    dps:
+      - id: 1
+        type: integer
+        unit: kWh
+        class: total_increasing
+        force: true
+        name: sensor
+        mapping:
+          - scale: 100
+  - entity: sensor
+    name: Total Reverse Energy
+    class: energy
+    # category: diagnostic
+    dps:
+      - id: 2
+        type: integer
+        unit: kWh
+        class: total_increasing
+        force: true
+        name: sensor
+        mapping:
+          - scale: 100
+  - entity: sensor
+    class: voltage
+    # category: diagnostic
+    dps:
+      - id: 6
+        optional: true
+        force: true
+        type: base64
+        unit: V
+        class: measurement
+        name: sensor
+        mask: "FFFF000000000000"
+        mapping:
+          - scale: 10
+  - entity: sensor
+    class: current
+    # category: diagnostic
+    dps:
+      - id: 6
+        optional: true
+        force: true
+        type: base64
+        unit: A
+        name: sensor
+        mask: "0000FFFFFF000000"
+        mapping:
+          - scale: 1000
+  - entity: sensor
+    class: power
+    # category: diagnostic
+    dps:
+      - id: 6
+        optional: true
+        force: true
+        type: base64
+        unit: kW
+        name: sensor
+        mask: "0000000000FFFFFF"
+        mapping:
+          - scale: 1000
+  - entity: sensor
+    name: Alarm Set 2
```

Comment:
This is likely an encoded binary field, so not useful as a sensor. Instead include it as an attribute of the primary entity.